iacm-plan.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"harness/client"
+	"harness/utils"
+	"os"
+	"path/filepath"
+
+	"github.com/fatih/color"
+	"github.com/hashicorp/go-slug"
+
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	orgIdFlag       = "org-id"
+	projectIdFlag   = "project-id"
+	workspaceIdFlag = "workspace-id"
+
+	folderPathWarningMsg   = "The workspace is configured with the folder path %s,\nHarness will upload the following directory and its contents: \n%s"
+	noFolderPathWarningMsg = "The workspace has no configured folder path,\nHarness will upload the following directory and its contents \n%s"
+	folderPathNotFoundErr  = "The folder path configured in the workspace %s does not exist in the current directory"
+	folderPathErr          = "An error occurred when trying to find the folder path in the current directory: %v"
+)
+
+type Client interface {
+	GetWorkspace(ctx context.Context, org string, project string, workspace string) (*client.Workspace, error)
+	CreateRemoteExecution(ctx context.Context, org, project, workspace string) (*client.RemoteExecution, error)
+	UploadRemoteExecution(ctx context.Context, org, project, workspace, id string, file []byte) (*client.RemoteExecution, error)
+	ExecuteRemoteExecution(ctx context.Context, org, project, workspace, id string) (*client.RemoteExecution, error)
+}
+
+type IacmCommand struct {
+	client  Client
+	account string
+	debug   bool
+}
+
+func NewIacmCommand(account string, client Client) IacmCommand {
+	return IacmCommand{
+		account: account,
+		client:  client,
+	}
+}
+
+func (c IacmCommand) executePlan(ctx context.Context, org string, project string, workspace string) error {
+	fmt.Printf("Fetching workspace %s...", workspace)
+	ws, err := c.client.GetWorkspace(ctx, org, project, workspace)
+	if err != nil {
+		fmt.Println(utils.GetColoredText(fmt.Sprintf("An error occurred when fetching the workspace: %v", err.Error()), color.FgRed))
+		return err
+	}
+
+	defaultPipeline, err := getDefaultPipeline(ws.DefaultPipelines)
+	if err != nil {
+		fmt.Println(utils.GetColoredText(err.Error(), color.FgRed))
+	}
+	fmt.Printf("Condigured default pipeline: %s...", defaultPipeline)
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	warning, err := validateWorkdingDirectory(wd, ws)
+	if err != nil {
+		fmt.Println(utils.GetColoredText(err.Error(), color.FgRed))
+		return err
+	}
+
+	fmt.Println(warning)
```

Comment:
Debug?

Reply:
no that gives the warning of which part of the fs the upload happens from, that warning messages is determined by validateWorkingDirectory